isis/tests/FunctionalTestsHicrop.cpp

```diff
@@ -0,0 +1,437 @@
+1.5        0.6      264289110.06
+    )"; 
+  
+  jitterPath = prefix.path() + "/jitter.txt";
+  QFile jitterFile(jitterPath); 
+  
+  if (jitterFile.open(QIODevice::WriteOnly)) {
+    QTextStream out(&jitterFile); 
+    out << jitter;
+    jitterFile.close(); 
+  }
+  else { 
+    FAIL() << "Failed to create Jitter file" << std::endl;
+  }
+
+  QString outCubeFileName = prefix.path() + "/outTemp.cub";
+  
+  QVector<QString> args = {"to="+outCubeFileName, "source=Jitter", "jitter="+jitterPath, "CK="+ckPath,
+  "SCLK="+sclkPath, "LSK="+lskPath};
+  
+  UserInterface options(APP_XML, args);
+  Pvl *logs = new Pvl();
+  try {
+     hicrop(testCube, options, logs);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to open image: " << e.what() << std::endl;
+  }
+ 
+  PvlGroup results = logs->findGroup("Results");
+  
+  EXPECT_EQ((int)results.findKeyword("InputLines"), 1056);
+  EXPECT_EQ((int)results.findKeyword("NumberOfLinesCropped"), 103);
+  EXPECT_EQ((int)results.findKeyword("OututStartingLine"), 1);
+  EXPECT_EQ((int)results.findKeyword("OututEndingLine"), 953);  
+  EXPECT_EQ((int)results.findKeyword("OututLineCount"), 953); 
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)results.findKeyword("OututStartTime"), "2008-05-17T09:37:24.7841228");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)results.findKeyword("OututStopTime"), "2008-05-17T09:37:24.8747174");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)results.findKeyword("OututStartClock"), "4/0895484264.57342"); 
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)results.findKeyword("OututStopClock"), "4/0895484264.63279"); 
+ 
+  Cube oCube(outCubeFileName); 
+  PvlGroup oCubeInstrument = testCube->label()->findObject("IsisCube").findGroup("Instrument");
+  
+  EXPECT_EQ(oCube.lineCount(), 953);
+  EXPECT_EQ(oCube.sampleCount(), 1204);
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)oCubeInstrument.findKeyword("StartTime"), "2008-05-17T09:37:24.7300819");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)oCubeInstrument.findKeyword("StopTime"), "2008-05-17T09:37:31.0666673");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)oCubeInstrument.findKeyword("SpacecraftClockStartCount"), "895484264:57342");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)oCubeInstrument.findKeyword("SpacecraftClockStopCount"), "895484272:12777");
+}
+
+
+TEST_F(MroCube, FunctionalTestHicropInstrumentError) {
+  setInstrument("-74999", "NOT_HIRISE", "MARS RECONNAISSANCE ORBITER");
+  
+  QTemporaryDir prefix;
+  QString outCubeFileName = prefix.path() + "/outTemp.cub";
+  
+  QVector<QString> args = {"to="+outCubeFileName, "source=Jitter", "jitter="+jitterPath, "CK="+ckPath,
+  "SCLK="+sclkPath, "LSK="+lskPath};
+  
+  UserInterface options(APP_XML, args);
+  Pvl *logs = new Pvl();
+
+  try {
+     hicrop(testCube, options, logs);
+  }
+  catch (IException &e) {
+    EXPECT_THAT(e.what(), HasSubstr("Input cube has invalid InstrumentId"));;
```

Comment:
Should we use the predefined functions in TestUtils to make sure the error message contains a substring?

Reply:
I don't think so, I prefer to use gtest tools as they were intended and not hide it behind functions calls. I feel like the EXPECT_THAT(HasSubStr) approach is simple and very explicit.